src/components/sectionList/listView/ManageListView.tsx

```diff
@@ -0,0 +1,110 @@
+}
+
+const toPath = (propertyDescriptor: { path: string }) => propertyDescriptor.path
+
+export const ManageListView = ({
+    onSaved,
+    children,
+}: ManageColumnsDialogProps) => {
+    const section = useModelSectionHandleOrThrow()
+    // ignore updates to saved-columns while selecting
+    const isTouched = useRef(false)
+
+    const { columns: savedColumns, query } = useModelListView()
+    const [pendingSelectedColumns, setPendingSelectedColumns] = useState<
+        string[]
+    >(() => savedColumns.map(toPath))
+    const { saveColumns, mutation } = useMutateModelListViews()
+
+    const columnsConfig = getColumnsForSection(section.name)
+
+    useEffect(() => {
+        // if savedColumns were to update while selecting (it shouldn't )
+        // make sure to not overwrite the selected columns
+        if (isTouched.current) {
+            return
+        }
+        setPendingSelectedColumns(savedColumns.map(toPath))
+    }, [savedColumns])
+
+    const handleSave = () => {
+        saveColumns(pendingSelectedColumns, {
+            onSuccess: () => onSaved(),
+        })
+    }
+
+    const handleSetDefault = () => {
+        setPendingSelectedColumns(columnsConfig.default.map(toPath))
+    }
+
+    const handleChange = ({ selected }: { selected: string[] }) => {
+        isTouched.current = true
+        setPendingSelectedColumns(selected)
+    }
+
+    const transferOptions = useMemo(
+        () =>
+            columnsConfig.available
+                .map((column) => ({
+                    label: column.label,
+                    value: column.path,
+                }))
+                .sort((a, b) => a.label.localeCompare(b.label)),
+        [columnsConfig.available]
+    )
+
+    return (
+        <>
+            <Transfer
+                height={'320px'}
+                enableOrderChange
+                leftHeader={
+                    <TransferHeader>
+                        {i18n.t('Available table columns')}
+                    </TransferHeader>
+                }
+                rightHeader={
+                    <TransferHeader>
+                        {i18n.t('Selected table columns')}
+                    </TransferHeader>
+                }
+                onChange={handleChange}
+                loading={query.isLoading}
+                loadingPicked={query.isLoading}
+                options={transferOptions}
+                selected={pendingSelectedColumns}
+            />
+            <Button
+                className={css.resetDefaultButton}
+                small
+                secondary
+                onClick={handleSetDefault}
+                disabled={mutation.isLoading}
+            >
+                {i18n.t('Reset to default columns')}
+            </Button>
+            {children({ handleSave, isSaving: mutation.isLoading })}
+        </>
+    )
+}
+
+const TransferHeader = ({ children }: React.PropsWithChildren) => (
+    <div className={css.transferHeader}>{children}</div>
+)
```

Comment:
I guess this is a style-preference - I prefer the main component to be the first thing in the file. Definition after usage of components doesnt really matter - since the `TransferHeader` is already defined when it's actually used and called.